modules/GKB/CommonUtils.pm

```diff
@@ -112,204 +114,259 @@ sub restore_database {
         push @species, get_all_species_in_entity($_) foreach (@{$entity->hasMember}, @{$entity->hasComponent});
     }
-    
+
     my %species;
     $species{$_->db_id} = $_ foreach @species;
     return values %species;
 }
 
 sub is_electronically_inferred {
-	my $instance = shift;
-
-	my $dba = $instance->dba();
-	my $db_name = $dba->db_name();
-	return 0 unless $db_name =~ /^test_reactome_\d+$/ || $db_name eq 'gk_current';
-
-	# first, let's check to see if the "magic" note is in the Created InstanceEdit for this object.
-	my $created = $instance->created->[0];
-	if ($created && $created->note && $created->note eq "inferred events based on ensembl compara")
-	{
-		return 1;
-	}
-	# If there's no note to indicate that the object was created during orthoinference,
-	# try the old logic.
-	if ($instance->is_a('Event')) {
-		return $instance->evidenceType->[0] && $instance->evidenceType->[0]->displayName =~ /electronic/i;
-	} elsif ($instance->is_a('PhysicalEntity')) {
-		# If manually inferred physical entities ever start using the inferredFrom slot, this logic
-		# can be augmented to check the event(s) to which the physical entity is/are attached
-		# to see if the event(s) are electronically inferred
-		return $instance->inferredFrom->[0] &&
-			$instance->inferredFrom->[0]->species->[0] &&
-			$instance->inferredFrom->[0]->species->[0]->displayName =~ /^Homo sapiens$/i;
-	} elsif ($instance->is_a('CatalystActivity')) {
-		return any { is_electronically_inferred($_)} @{$instance->reverse_attribute_value('catalystActivity')};
-	} elsif ($instance->is_a('Regulation')) {
-		return any { is_electronically_inferred($_)} @{$instance->reverse_attribute_value('regulatedBy')};
-	}
+    my $instance = shift;
+
+    my $dba = $instance->dba();
+    my $db_name = $dba->db_name();
+    return 0 unless $db_name =~ /^test_reactome_\d+$/ || $db_name eq 'gk_current';
+
+    # first, let's check to see if the "magic" note is in the Created InstanceEdit for this object.
+    my $created = $instance->created->[0];
+    if ($created && $created->note && $created->note eq "inferred events based on ensembl compara") {
+        return 1;
+    }
+    # If there's no note to indicate that the object was created during orthoinference,
+    # try the old logic.
+    if ($instance->is_a('Event')) {
+        return $instance->evidenceType->[0] && $instance->evidenceType->[0]->displayName =~ /electronic/i;
+    } elsif ($instance->is_a('PhysicalEntity')) {
+        return !in_gk_central($instance) && has_human_source_instance($instance);
+    } elsif ($instance->is_a('CatalystActivity')) {
+        return any { is_electronically_inferred($_)} @{$instance->reverse_attribute_value('catalystActivity')};
+    } elsif ($instance->is_a('Regulation')) {
+        # For backwards compatability when regulations contained the reaction like event (RLE) they regulated
+        # As of Reactome version 65, RLEs have contained their regulation instances in the 'regulatedBy' attribute
```

Comment:
This is to allow queries to previous versions of release databases.  The regulations in those databases won't ever be updated I don't think because the data model they have doesn't support it.

Reply:
Ah, ok.